TypeCobol/Compiler/Symbols/SymbolExpander.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TypeCobol.Compiler.Types;
+using Type = System.Type;
+
+namespace TypeCobol.Compiler.Symbols
+{
+    /// <summary>
+    /// Symbol Expander
+    /// </summary>
+    public class SymbolExpander : Symbol.AbstractSymbolVisitor<Symbol, Symbol>
+    {
+        public ProgramSymbol Program
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The Type Expander to use.
+        /// </summary>
+        public TypedefExpander TypeExpander
+        {
+            get;
+            internal set;
+        }
+
+        /// <summary>
+        /// This will be an indication if the Symbol shall be level renumbered.
+        /// If this Symbol Expander is used for several symbol, it will be necessary for each
+        /// Acceptation to reset this variable to false before. 
+        /// </summary>
+        public bool ShallLevelRenumber
```

Comment:
Yes because, the expanding mechanism does not renumber variable level, thus for instance a codegen must explicitly use LevelRenumber do do that, so it can first check if it is necessary to do that.